test/e2e/set/scyllacluster/scyllacluster_pv.go

```diff
@@ -178,6 +180,10 @@ var _ = g.Describe("ScyllaCluster Orphaned PV controller", func() {
 							},
 							TerminationGracePeriodSeconds: pointer.Ptr(int64(1)),
 							RestartPolicy:                 corev1.RestartPolicyNever,
+							Tolerations:                   slices.Clone(sc.Spec.Datacenter.Racks[0].Placement.Tolerations),
+							NodeSelector: map[string]string{
+								"scylla.scylladb.com/node-type": "scylla",
+							},
```

Comment:
why wasn't this there before?

also if you want the same node, then set nodeName and avoid the scheduler

Reply:
> why wasn't this there before?

Because we were running everything in the same pool due to the misconfigured taints. Now this workload needs to run in scylla-dedicated pool, which is normally dictated by our workloads using `scylladb-local-xfs` StorageClass. This one does not have it, so it needs to be set explicitly. Otherwise it runs in the generic worker pool, and the PodAffinity on ScyllaCluster also causes it to run there, which in turn causes it to error due to the worker nodes not having `aio-max-nr` set up.

Reply:
ok, I thought this was doing something slightly different

Reply:
shouldn't this technically be on the scyllacluster and just copied here?

Reply:
we could set the nodeAffinity on scylla clusters, yes, since we already set the nodeSelector on NodeConfigs either way. But in that case, given we just made the effort to unify taints/tolerations, shouldn't we set it in all our examples/manifests/fixtures?

Reply:
Or do you mean just in this particular test?

Reply:
(fixed for this test's scope)

Reply:
> But in that case, given we just made the effort to unify taints/tolerations, shouldn't we set it in all our examples/manifests/fixtures?

yep, everywhere (the fixture). These ScyllaClusters actually do target this node selector. (It "accidentally" works without saying it because of the storage affinity.)

Reply:
Ok, I'll send a PR setting node affinity on all ScyllaClusters and remove it from this test

Reply:
I made it a part of this PR since otherwise we wouldn't be able to merge either without overriding CI.

Reply:
maybe https://github.com/scylladb/scylla-operator/pull/2275 would unblock it but it's fine with me